src/components/ListItem.jsx

```diff
@@ -1,5 +1,56 @@
 import './ListItem.css';
+import { updateItem } from '../api';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+const currentTime = new Date();
+
+export function ListItem({ name, items, listToken }) {
+	const { isChecked, id, totalPurchases, dateLastPurchased } = items;
+
+	const currentDate = new Date();
+
+	if (isChecked) {
+		const currentTimeToSec = currentDate.getTime() / 1000;
+		const lastPurchaseToMilSec = dateLastPurchased.seconds * 1000;
+		const timeElapsed = currentTimeToSec - lastPurchaseToMilSec;
+
+		const timeToUncheck = 86400;
+		if (timeElapsed >= timeToUncheck) {
+			const itemData = {
+				isChecked: false,
+			};
+			updateItem(listToken, id, itemData);
+		}
+	}
+
+	const handleCheckbox = () => {
+		if (isChecked) {
+			const itemData = {
+				isChecked: false,
+			};
+			updateItem(listToken, id, itemData);
```

Comment:
The user shouldn't be able to uncheck a checked checkbox. Actually the checkbox it would even better if the checkbox were disabled after it's clicked by the user so they can't click on it again :slightly_smiling_face:

Reply:
that is another nice idea. Never thought of this. 
@Mama-simba, what do you think?

Reply:
thanks once again! I personally got confused by the fact of:
- checkbox is false by default
- checkbox is true when the user clicks on it
- checbox is false after 24h
but yes, @Terieyenike it is a good idea :)